src/Vector4_TEST.cc

```diff
@@ -61,6 +61,16 @@ TEST(Vector4dTest, Vector4d)
   v.Set(2, 4, 6, 8);
   EXPECT_EQ(v, math::Vector4d(2, 4, 6, 8));
 
+  // ::DotProd
+  EXPECT_TRUE(math::equal(60.0, v.Dot(math::Vector4d(1, 2, 3, 4)), 1e-2));
+
+  // ::AbsDotProd
+  v1.Set(-1, -2, -3, -4);
+  EXPECT_TRUE(math::equal(60.0, v.AbsDot(v1), 1e-2));
+
+  // ::GetAbs
+  EXPECT_TRUE(v1.Abs() == math::Vector4d(1, 2, 3, 4));
```

Comment:
Maybe a test of `Abs` with a vector with positive values (for completeness).

Reply:
Great! Added in the last commit.